apps/examples/testcase/ta_tc/task_manager/itc/itc_task_manager_main.c

```diff
@@ -0,0 +1,286 @@
+static task_info_t *second_sample_info;
+
+static void test_handler(void *info)
+{
+	flag = !strncmp(info, TM_SAMPLE_MSG, strlen(TM_SAMPLE_MSG));
+}
+
+int itc_sample_main(int argc, char *argv[]) {
+	int ret;
+
+	tm_noperm_handle = task_manager_register(TM_NOPERM_NAME,
+	TM_TASK_PERMISSION_DEDICATE, 100);
+
+	ret =task_manager_set_unicast_cb(test_handler);
+	if (ret != OK) {
+		printf("ERROR : fail to set handler\n");
+	}
+	while (1)
+		;	// This will be dead at itc_task_manager_terminate_p
+}
+
+/**
+ * @testcase         itc_task_manager_clean_info_twice
+ * @brief            Get task_manager_getinfo_with_handle twice
+ * @scenario         Get task_manager_getinfo_with_handle twice
+ * @apicovered       task_manager_getinfo_with_handle
+ * @precondition     NA
+ * @postcondition    NA
+ */
+static void itc_task_manager_clean_info_twice(void) {
+	sample_info = (task_info_t *) task_manager_getinfo_with_handle(
+			tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_NEQ("task_manager_clean_info", (task_info_t * )&sample_info, NULL);
+
+	task_manager_clean_info(&sample_info);
+	TC_ASSERT_NEQ("task_manager_clean_info", (task_info_t * )&sample_info, NULL);
+
+	TC_SUCCESS_RESULT()
+	;
+}
+
+
+/**
+ * @testcase         itc_task_manager_register_n
+ * @brief            Register task manager with invalid parameter
+ * @scenario         Register task manager with invalid parameter
+ * @apicovered       task_manager_register
+ * @precondition     NA
+ * @postcondition    NA
+ */
+
+static void itc_task_manager_register_n(void) {
+	int ret;
+	ret = task_manager_register(NULL, TM_TASK_PERMISSION_DEDICATE,
+	TM_NO_RESPONSE);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_INVALID_PERMISSION,
+	TM_NO_RESPONSE);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_TASK_PERMISSION_GROUP,
+	TM_INVALID_TIMEOUT);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME,
+	TM_INVALID_UPPER_RANGE_PERMISSION, TM_INVALID_TIMEOUT);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_TASK_PERMISSION_GROUP,
+	TM_NO_RESPONSE);
+	TC_ASSERT_NEQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	TC_SUCCESS_RESULT()
+	;
+}
+/**
+ * @testcase         itc_task_manager_register_p
+ * @brief            Register task manager with valid parameter
+ * @scenario         Register task manager with valid parameter
+ * @apicovered       task_manager_register
+ * @precondition     NA
+ * @postcondition    NA
+ */
+static void itc_task_manager_register_p(void) {
+
+	tm_sample_handle = task_manager_register("invalid",
+	TM_TASK_PERMISSION_DEDICATE, TM_RESPONSE_WAIT_INF);
+
+	TC_ASSERT_EQ("task_manager_register_p", tm_sample_handle,
+			TM_OPERATION_FAIL);
+	tm_sample_handle = task_manager_register(TM_SAMPLE_NAME,
```

Comment:
You set the builtin task name to itc_tm_sample, but the macro of TM_SAMPLE_NAME is tm_sample. So I think this TC cannot work properly. Please check this.

Reply:
changes done as per your suggestion.After changes, I also checked the tc and is executing with PASS